test/tools/router_check/router.cc

```diff
@@ -0,0 +1,206 @@
+#include "router.h"
+
+bool RouterCheckTool::create(const std::string& filename_json) {
+  // Throws expection if json file not correctly written
+  Json::ObjectPtr loader = Json::Factory::LoadFromFile(filename_json);
+  if (loader->empty()) {
+    log_debug("JSON file load failed.");
+    return false;
+  }
+
+  // Throws expection if json file not correctly written
+  loader->validateSchema(Json::Schema::ROUTE_CONFIGURATION_SCHEMA);
+
+  // Router configuration
+  config_ = new Router::ConfigImpl(*loader, runtime_, cm_, false);
```

Comment:
+1, please get into the habit of always writing leak free (and exception safe) code even in utilities like this. unique_ptr is the way to go.